src/backend/cdb/dispatcher/cdbdisp_query.c

```diff
@@ -871,6 +872,8 @@ buildGpQueryString(DispatchCommandQueryParms *pQueryParms,
 	int			total_query_len;
 	char	   *shared_query,
 			   *pos;
+	int			cnt = 0,
```

Comment:
May be it is more clear to define `cnt` variable here, but initialize it with 0 near the added loop.